tests/ga/extension_emulator.py

```diff
@@ -197,6 +199,27 @@ def _extend_func(func):
         def wrapped_func(cmd, *args, **kwargs):
             return_value = func(cmd, *args, **kwargs)
 
+            config_dir = os.path.join(os.path.dirname(cmd), "config")
+            
+            regex = r'{directory}{sep}(?P<seq>{sequence})\.settings'.format(
+                directory=config_dir, sep=os.path.sep, sequence=r'[0-9]+'
+            )
+
+            seq = 0
+            for config_file in map(lambda filename: os.path.join(config_dir, filename), os.listdir(config_dir)):
+                if not os.path.isfile(config_file):
+                    continue
+
+                match = re.fullmatch(regex, config_file)
+                if not match:
+                    continue
+
+                if seq < int(match.group("seq")):
+                    seq = int(match.group("seq"))
+
+            fileutil.write_file(os.path.join(os.path.dirname(cmd), "status", "{seq}.status".format(seq=seq)),
+                json.dumps([{ "status": {"status": "success"} }]))
```

Comment:
NIT: If you always report success, there might be a scenario where the status being reported is success but the process failed with some error. Might be worth making this a bit more strict to report success only if return_code == 0 else "failed" or something.

Reply:
did we decide not to address this comment?

Reply:
I looked into this, but forgot to document what I found here.

Basically, the object written to file here is only ever sent to the `HTTP PUT /status` endpoint when there is not some sort of failure. In other words, it doesn't seem possible that `return_code == 0` and that status reporting success is conveyed at the same time.

A perfect example of this is the test `tests.ga.test_extension.test_extension_error_should_be_raised_when_continue_on_update_failure_is_true_on_disable_and_update_failure`, wherein update fails and the only status reported for that extension is the exit code we outputted (instead of the `status: success` stuff). This is verified by an `assertTrue` within the test, and I went back and stepped through the code, and it is in fact the case.

Reply:
>wherein update fails and the only status reported for that extension is the exit code we outputted (instead of the status: success stuff).

There are multiple levels to our status report, we have the HandlerStatus and ExtensionSettings settings. The HandlerStatus failure is set independently of the extension status. I think since you aren't checking the individual setting status of any extension you will probably not see this. Even though this might not be needed scenario for your changes, I'm wondering if we'd need this later (since you are creating a new framework).

Reply:
Ah, that sounds like it very well might be the case. Do you mind pointing me to where in `exthandlers.py` we do the reporting of the extension status (i.e., the status always being reported as `success` here)?

I'd love to take a look at it. If this other status is submitted somewhere and has a use-case, I agree the code here should be flexible; even though I don't think it would be particularly tricky to add the if after the fact (when/if we run into the need for it), I don't have any problems with adding it in this PR if there is a use-case we'll want to test for.

Reply:
This function `report_ext_status` in exthandlers.py does that parsing. You can follow the thread here

In your test code, in line: 121 in `extension_emulator.py`, I think you're getting the handler_status. If you debug the whole `aggregateState` that you pull from the status blob, you should see the extension status too (not just the handler status).

I can walk the debug with you if you want if it gets too confusing, let me know

Reply:
>If this other status is submitted somewhere and has a use-case

There might be instances where we need this, like if we want to verify the extension status for a specific sequence no for that extension. Given that you are creating a cool new framework, this would definitely be helpful